### DISCUSSION TITLE
Jobs running multiples times after each deployment to the azure services

### DISCUSSION THREAD

**Discussion by hello2pankaj** [quartznet/quartznet#2348] [Q&A]
**Describe the bug**

I am using Quartz.Extensions.Hostin 3.8.1 scheduling package in .netcore6 web api application. App deployment environment is azure appservice (2 instances). It works fine and as expected. Only issue i am facing is, after every deployment , the jobs are run 2-5 times in quick succession. The time difference between the runs ranges from few seconds to couple of minutes. After this very first trigger, other trigger happens at defined intervals properly. Below is my quartz config. Any idea.

For example, my job is scheduled to run at 1 hour interval. But immediately after deployment , it run 4 times (green) and then properly at 1 hour interval (yellow)

![image](https://github.com/quartznet/quartznet/assets/15063036/efd38d5e-950b-4e31-b115-d4ff5622f613)

**Version used**
Quartz.Extensions.Hostin 3.8.1.

**To Reproduce**

```
public static void ConfigureQuartzJobs(this WebApplicationBuilder builder)
{
	var services = builder.Services;
	services.AddQuartzHostedService(opt => opt.WaitForJobsToComplete = true);
	services.AddQuartz(q =>
	{
		q.SchedulerId = "SomeId";
		q.SchedulerName = "SomeName";
		q.UsePersistentStore(s =>
				{
				  s.UseProperties = true;
				  s.UseClustering();
				  s.UseSqlServer(connectionString);
				  s.UseNewtonsoftJsonSerializer();
				});
				q.AddJob<job1>(jobkey1)
				 .AddTrigger(trigger => trigger
						   .ForJob(jobkey1)
						   .WithIdentity(tr1)
						   .StartAt(DateTimeOffset.UtcNow.AddSeconds(5))
						   .WithSimpleSchedule(schedule => schedule.WithIntervalInMinutes(5).RepeatForever()));
				});
	}
}
```

Job1 class

```
[DisallowConcurrentExecution]
public class Job1 : IJob
{
	public async Task Execute(IJobExecutionContext context)
	{
		try
		{
			//Call logic classes
		}
		catch (Exception ex)
		{
			await Console.Out.WriteLineAsync($"{context.FireTimeUtc}, {ex}. FireInstanceId={context.FireInstanceId}");
		}
	}
}
```

**Expected behavior**
After deployment, Job either should run 1 time or not at all so that the next trigger fires it.

**Additional context**
I have tried with below configs
		
		q.UseDefaultThreadPool(x => x.MaxConcurrency = 5);
		q.MisfireThreshold = TimeSpan.FromSeconds(60);
		
		
		 trigger .StartAt(DateTimeOffset.UtcNow.AddSeconds(5))
		 trigger.StartNow()
		 Without any start time option as well

**Comment by lahma** [comment]
`StartAt(DateTimeOffset.UtcNow.AddSeconds(5))` look like relative to each application start and is not fixed , if `tr1` is not a fixed name there will be multiple triggers too.

**Reply by hello2pankaj** [reply]
In total , i have 4 jobs. All these jobs have same issue on deployment. After that, they run as expected, Below is my latest config as i am doing trial and error. Keeping or Removing StartAt OR StartNow Or no start config have no affect.  Also tr1 is a fixed name. All the triggers have distinct identity in actual implementation.  However,  even if i remove other jobs and keep only 1 job and trigger, behavior is same, It runs multiple times after each deployment before running properly at defined intervals.

```
q.AddJob<job1>(jobkey1)
 .AddTrigger(trigger => trigger
		   .ForJob(jobkey1)
		   .WithIdentity("tr1")
		   .WithSimpleSchedule(schedule => schedule.WithIntervalInMinutes(5).RepeatForever()));
		   

q.AddJob<job2>(jobkey2)
 .AddTrigger(trigger => trigger
		   .ForJob(jobkey3)
		   .WithIdentity("tr2")
		   .WithSimpleSchedule(schedule => schedule.WithIntervalInMinutes(5).RepeatForever()));
		   

q.AddJob<job3>(jobkey3)
 .AddTrigger(trigger => trigger
		   .ForJob(jobkey3)
		   .WithIdentity("tr3")
		   .WithSimpleSchedule(schedule => schedule.WithIntervalInMinutes(5).RepeatForever()));
		   

q.AddJob<job4>(jobkey4)
 .AddTrigger(trigger => trigger
		   .ForJob(jobkey4)
		   .WithIdentity("tr4")
		   .WithSimpleSchedule(schedule => schedule.WithIntervalInMinutes(60).RepeatForever()));
```

**Reply by lahma** [reply]
Without start time it defaults to "now" which is again not fixed moment. If you are using persistent store you should only add triggers if they are missing and preferably use a fixed start time for them.

**Reply by hello2pankaj** [reply]
Ok. I am not sure and also couldn't find in documentation how to check for existing triggers in below code structure. Would you please guide me?

```
public static void ConfigureQuartzJobs(this WebApplicationBuilder builder)
{
	var services = builder.Services;
	services.AddQuartzHostedService(opt => opt.WaitForJobsToComplete = true);
	services.AddQuartz(q =>
	{
		q.SchedulerId = "SomeId";
		q.SchedulerName = "SomeName";
		q.UsePersistentStore(s =>
				{
				  s.UseProperties = true;
				  s.UseClustering();
				  s.UseSqlServer(connectionString);
				  s.UseNewtonsoftJsonSerializer();
				});
				q.AddJob<job1>(jobkey1)
				 .AddTrigger(trigger => trigger
						   .ForJob(jobkey1)
						   .WithIdentity("tr1")
						   .StartAt(DateTimeOffset.UtcNow.AddSeconds(5))
						   .WithSimpleSchedule(schedule => schedule.WithIntervalInMinutes(5).RepeatForever()));
				});
	}
}
```
